solidity/contracts/PhasedEscrow.sol

```diff
@@ -70,26 +70,127 @@ contract PhasedEscrow is Ownable {
     }
 }
 
-interface ICurveRewards {
-    function notifyRewardAmount(uint256 amount) external;
+/// @title BatchedPhasedEscrow
+/// @notice A token holder contract allowing contract owner to approve a set of
+///         beneficiaries of tokens held by the contract, to appoint a separate
+///         drawee role, and allowing that drawee to withdraw tokens to approved
+///         beneficiaries in phases.
+contract BatchedPhasedEscrow is Ownable {
+    using SafeERC20 for IERC20;
+
+    event BeneficiaryApproved(address beneficiary);
+    event TokensWithdrawn(address beneficiary, uint256 amount);
+    event DraweeRoleTransferred(address oldDrawee, address newDrawee);
```

Comment:
Do we want to add some indexing?

```suggestion
    event BeneficiaryApproved(address indexed beneficiary);
    event TokensWithdrawn(address indexed beneficiary, uint256 amount);
    event DraweeRoleTransferred(address indexed oldDrawee, address indexed newDrawee);
```

Reply:
I am not sure, we'll approve 4 beneficiaries and transfer the drawee role one time. I'd be surprised if we had more than 10 those events emitted in total.